### PR TITLE
Pepe

### PR DESCRIPTION
**Pull request by JoseAntRodriguez**
Models and population script created. As far as I'm aware, they work properly. If not working you might have to delete the migrations folder and the database file, and the run 'python manage.py makemigrations gliocas_app', 'python manage.py migrate' and then the population script

### CODE REVIEW

**Review by theycallmepeter — APPROVED** [review]
Seems good.

**Review by Hodobox — COMMENTED** [review]
I think ideally this should generate an 'empty'  model at each level - e.g. empty subject, empty course, so we can check how that would look on our website (i.e. 'no subjects present' correctly displayed)

File: Gliocas/gliocas_app/models.py [hunk 1]
```diff
@@ -1,42 +1,92 @@
...
     slug = models.SlugField(unique=True)
-    # other attributes
 
     def save(self, *args, **kwargs):
-        self.slug = slugify(self.title)
-        super(Question,self).save(*args, **kwargs)
+        self.slug = slugify(self.pk)
+        super(Question, self).save(*args, **kwargs)
+    
+    def __str__(self):
```
Comment (Hodobox) [inline Gliocas/gliocas_app/models.py:60]: nonexistent field

File: Gliocas/gliocas_app/models.py [hunk 2]
```diff
@@ -1,42 +1,92 @@
...
+    textLength = 32768
+    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+    poster = models.ForeignKey(User,
+                               on_delete=models.SET(get_sentinel_user))
+    text = models.TextField(max_length=textLength)
+    date = models.DateTimeField(null=True)
+    upvotes = models.IntegerField(default=0)
+    
+    def __str__(self):
+        return self.name
```
Comment (Hodobox) [inline Gliocas/gliocas_app/models.py:73]: nonexistent field

File: Gliocas/gliocas_app/models.py [hunk 3]
```diff
@@ -1,42 +1,92 @@
...
+    poster = models.ForeignKey(User,
+                               on_delete=models.SET(get_sentinel_user))
+    text = models.TextField(max_length=textLength)
+    date = models.DateTimeField(null=True)
+    upvotes = models.IntegerField(default=0)
+    
     def __str__(self):
-        return self.title
\ No newline at end of file
+        return self.name
```
Comment (Hodobox) [inline Gliocas/gliocas_app/models.py:85]: again, nonexistent field

File: Gliocas/gliocas_app/models.py [hunk 4]
```diff
@@ -1,42 +1,92 @@
...
+    poster = models.ForeignKey(User,
+                               on_delete=models.SET(get_sentinel_user))
+    text = models.TextField(max_length=textLength)
+    date = models.DateTimeField(null=True)
+    upvotes = models.IntegerField(default=0)
+    
+    def __str__(self):
+        return self.name
 
+class Reply(models.Model):
```
Comment (Hodobox) [inline Gliocas/gliocas_app/models.py:75]: can you add a plural-noun override, so we don't get 'Replys'?

File: Gliocas/populate.py [hunk 5]
```diff
@@ -1,59 +1,187 @@
+import datetime
 import os
-
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'Gliocas.settings')
 
 import django
 django.setup()
+from django.contrib.auth.models import User
 from gliocas_app.models import Subject, Course, Question
```
Comment (Hodobox) [inline Gliocas/populate.py:8]: one line please for imports from gliocas_app

File: Gliocas/populate.py [hunk 6]
```diff
@@ -1,59 +1,187 @@
...
+         "text" : ("I took this subject and I have no idea about what"
+                   " we are doing. Can anyone tell me, to begin with,"
+                   " who is this Robert Burns?"),
+         "poster" : users["proudScottish"],
+         "views" : 17,
+         "upvotes" : 5,
+         "answers" : {}},
+        ]
+    genderHistory_questions = []
+    spanishHistory_questions = []
```
Comment (Hodobox) [inline Gliocas/populate.py:99]: would be nice to have a subject with no courses, to see how that would display.

File: Gliocas/populate.py [hunk 7]
```diff
@@ -1,59 +1,187 @@
...
+                print("\t\t", "upvotes", q.upvotes)
+                for a in Answer.objects.filter(question=q):
+                    print("\t\t\t", "text", a.text)
+                    print("\t\t\t", "poster", a.poster)
+                    print("\t\t\t", "date", a.date)
+                    print("\t\t\t", "upvotes", a.upvotes)
+
+
+def add_user(name):
+    user = User.objects.get_or_create(username=name,
```
Comment (Hodobox) [inline Gliocas/populate.py:149]: can the password be something easily typable and memorable? Most of us don't have this symbol on their keyboard. 'password' should be good enough for test database, no?

File: Gliocas/gliocas_app/models.py (hunk 1 above)
Comment (JoseAntRodriguez) [reply Gliocas/gliocas_app/models.py:60]: pk is actually the primary key, in this case an autogenerated integer

**Comment by JoseAntRodriguez** [comment]
All of the suggestions are implemented now

**Pull request merged**